src/Vixen.Modules/Editor/TimedSequenceEditor/TimedSequenceEditorForm.cs

```diff
@@ -652,6 +652,7 @@ protected override void Dispose(bool disposing)
 				//Dispose any nodes that where removed
 				if (!_effectNodeToElement.ContainsKey(node))
 				{
+					node.Effect.Removing(); // Notify Effect to clean up external resources, like cache folders

```

Comment:
I think this gets called anytime the editor is closed, thus the cached thumbnails will still be removed after each editing session. I thought part of this was to persist outside of an editing session, but maybe I am missing something.

Reply:
When I was tracing through what happens when deleting an Effect, I found that it's added to the _removedNodes list in the RemoveEffectNodeAndElement method. The only other place I found where that gets used is in the Editor's Dispose method. It looks at each node in the _removedNodes list and so long as it isn't in the _effectNodeToElement list (likely from the undo case), it calls that Effect's Dispose. I created and added the Removing() method as a way to let an Effect know that it has been deleted and not just Disposed. Just after this foreach is another that calls Dispose on all of the non-deleted Effects.

The other half is in Video.cs where I deleted the Dispose method and created the Removing method to call RemoveTempFiles instead. This way the cache folder is only removed if the effect was deleted when the editor is closed.